haskell/test/Test/Ligo/Common.hs

```diff
@@ -0,0 +1,73 @@
+-- SPDX-FileCopyrightText: 2021 Tezos Commons
+-- SPDX-License-Identifier: LicenseRef-MIT-TC
+--
+module Test.Ligo.Common
+  ( IsProposalArgument(..)
+  , VariantExtraHasField(..)
+  , TestableVariant(..)
```

Comment:
```suggestion
  , TestableVariant(toProposalMetadata, getVariantStorageRPC)
```
This probably fixes the weeder, can you try it.

Reply:
On the side note, is it just me or the test compilation is noticeably slower? Especially during `RegistryDAO.Tests.**` tests

Reply:
It is not you. The compilation is now a lot slower due to the constraints added to the registry/treasury tests to use them for testing Lambda versions of the same.

Reply:
>This probably fixes the weeder, can you try it?

It does not appear possible to only export those methods, because the other ones are used in other modules.

Reply:
>This probably fixes the weeder, can you try it?

It does not appear possible to only export those methods, because the other ones are used in other modules.

Reply:
> It does not appear possible to only export those methods, because the other ones are used in other modules.

Right, I guess we can update weeder.yaml to ignore it, similarly to the one in `morley`: https://gitlab.com/morley-framework/morley/-/blob/master/.weeder.yaml#L23

> It is not you. The compilation is now a lot slower due to the constraints added to the registry/treasury tests to use them for testing Lambda versions of the same.

I think this be a huge problem, we should try to find way to reduce the compilation time.

Reply:
> I guess we can update weeder.yaml to ignore it,

Sure, but I would like to figure why is those considered as weeds first..

I had tried to address it by splitting the registryDAO tests into many different modules, each with a single test. But it only helps with incremental compilation, and not build-from-scratch that CI does.

Another alternative I can think of is to duplicate the entire test modules for registry and treasury. Maybe Haskell has got parameteric modules, we could have made good use of that here...